component/portal/src/main/java/org/exoplatform/portal/rest/UserRestResourcesV1.java

```diff
@@ -222,7 +222,11 @@ public Response createUser(@Context HttpServletRequest request, UserRestEntity u
     user.setFirstName(firstName);
     user.setLastName(lastName);
     user.setPassword(password);
-    organizationService.getUserHandler().createUser(user, true);
+    try {
+       organizationService.getUserHandler().createUser(user, true);
+    } catch (ObjectAlreadyExistsException objectAlreadyExistsException) {
```

Comment:
I think you dont have to create a new exception here. Catch the RuntimeException, and check the error message. If the error message is "Unable to create a previously deleted user", then you are in the case username already exists

Reply:
I see that it is not logical to throw RuntimeException in this case, again this exception returns status 500 and I cannot catch the error and with ObjectAlreadyExistsException I have a satuts 400 and i can catch the error message

Reply:
i think creating a new Exception class could help handling the right error specially from the front side using the right code error, he can throw 400 or 406 status error code in this case.
@rdenarie wdyt ?

Reply:
Ok to keep the new Exception class

Reply:
@rdenarie The idea is to use this Exception class and remove the one here https://github.com/Meeds-io/social/blob/develop/component/api/src/main/java/org/exoplatform/social/common/ObjectAlreadyExistsException.java
But I didn't want to we make this PR bigger and change in many places of the product. Maybe we could do it in another PR WDYT ?

Reply:
can we keep this exception and modify the social one to extends this one ? 
And mark it as deprecated ?
So, all classes using social class will continue to work, and we can start to use this class instead ?

Reply:
Seems good to me :+1:

Reply:
@abenmbarek can you make the mdification in social module  ? Thanks

Reply:
no problem

Reply:
@abenmbarek Im sorry, the message was for @Noussairhamrit which is the author of the PR :)

Reply:
ah it's okey, i'm here if he needs some help ;)